### ISSUE TITLE
Delta function

### ISSUE THREAD

**Issue by giorgoskallitsis** [ntua-cvsp-lab/DSP_LabSupport#20]
Καλησπέρα, αν και έχει ξαναρωτηθεί...στο ερώτημα 2.2.α, προσπάθησα να γράψω τις συναρτήσεις δέλτα με τη χρήση της συνάρτησης scipy.signal.unit_impulse, αλλά διαπίστωσα πως σαν δεύτερο όρισμα, όπου είναι το σημείο στο οποίο θα λάβει την τιμή 1,  απαιτείται μόνο integer. Πως θα εισάγουμε την τιμή 0.625 που ζητείται στην άσκηση?

**Comment by cgaroufis** [comment]
Καλησπέρα,

Για να βρεις σε ποια θέση ο παλμός του διανύσματος που σου επιστρέφει η unit_impulse() θα παίρνει την τιμή 1, θα πρέπει να χρησιμοποιήσεις τη συχνότητα δειγματοληψίας του σήματός που δίνεται (1000 Hz), για να βρεις την περίοδο δειγματοληψίας Τs και από εκεί, το δείκτη n από την αντιστοίχιση t (συνεχές) <-> nTs (διακριτό).

Καλή συνέχεια,
Χρήστος

**Comment by giorgoskallitsis** [comment]
Καλησπέρα, στα ερωτήματα 2.1α και 2.2α εγω σκέφτηκα να γραψουμε πρώτα τις συναρτήσεις στο συνεχές χρόνο και έπειτα να τις δειγματολειπτησουμε με την εντολή sp.signal.resample. Είναι λάθος η σκέψη μου?

**Comment by cgaroufis** [comment]
Η sp.signal.resample απ'όσο ξέρω μπορεί να χρησιμοποιηθεί για την επαναδειγματοληψία διακριτών σημάτων, καθώς το 1ο ορισμα που δέχεται πρέπει να είναι τύπου np.array η συναφούς (array-like στο documentation). 

Γενικά, ο ορισμός explicitly συνεχών εκφράσεων σε python μπορεί να γίνει μέσω μόνο συμβολικών πακέτων/βιβλιοθηκών (όπως για παράδειγμα η [sympy](https://docs.sympy.org/latest/index.html)). Παρέχεται δυνατότητα δειγματοληψίας τους και αποτίμησής τους σε συγκεκριμένα πεδία τιμών [μέσω της numpy](https://docs.sympy.org/latest/modules/numeric-computation.html), αλλά για τους σκοπούς της άσκησης είναι σημαντικά απλούστερος ο ορισμός των συναρτήσεων απευθείας σε ένα διακριτό χώρο (μέσω της numpy.linspace(), βλ. παράδειγμα στο Audio_Analysis_Intro.ipynb).

Χρήστος

**Comment by giorgoskallitsis** [comment]
Αρα θα πρεπει και στα δυο ερωτηματα να θέσω όπου t το n*Ts για να ορίσω τις συναρτήσεις και να επεξεργαστώ τα σήματα?

**Comment by cgaroufis** [comment]
Πρακτικα ναι, ορίζοντας το n ως linspace ανάμεσα στο 0 και στο 2 με Fs σημεία ανά δευτερόλεπτο.

**Comment by giorgoskallitsis** [comment]
Ναι αυτο το εχω ηδη κανει, απλως ρωταω αν πχ τον ορο με το συνημιτονο cos(2π90t) θα τον γραψουμε στην python ως cos(2π90n) ή ως cos(2π90*n*Ts)? Και αντιστοιχα τους ορους με τις συναρτησεις δ.

**Comment by cgaroufis** [comment]
Εξαρτάται από το πως ορίζεις το n. Αν το έχεις ορίσει από το 0 έως 2, ως cos(2π90n). Αν το εχεις ορίσει ως 0, 1, ..., ως cos(2π90nTs). Απλά, το k-οστό δείγμα σου (το οποίο και παίρνει ως όρισμα η unit_impulse της scipy) θα αντιστοιχεί στη χρονική στιγμή kTs.

**Comment by giorgoskallitsis** [comment]
Εγω ορισα το n να πηγαινει απο 0 εως 2. Αρα, μου λετε, σε αυτην την περιπτωση, να βαλω το cos ως cos(2π90n). Αλλα σε αυτην την περιπτωση, η δ θα γραφει ως δ(n-0.625), πραγμα το οποίο ειναι αδυνατο,αφου η unit_impulse δεχεται για ορισμα μονο ακεραιους. Αρα, τι να κανω?

**Comment by cgaroufis** [comment]
Όχι, επειδή η unit_impulse χρησιμοποιεί ως όρισμα το δείγμα (625) στο οποίο αντιστοιχεί το δ(t-0.625), και οχι τη χρονική στιγμή (0.625)

**Comment by giorgoskallitsis** [comment]
Τωρα το καταλαβα, ευχαριστω πολυ!
